python/pyspark/sql/streaming.py

```diff
@@ -276,9 +276,9 @@ def resetTerminated(self):
 
 class DataStreamReader(OptionUtils):
     """
-    Interface used to load a streaming :class:`DataFrame` from external storage systems
-    (e.g. file systems, key-value stores, etc). Use :func:`spark.readStream`
-    to access this.
+    Interface used to load a streaming :class:`DataFrame <pyspark.sql.DataFrame>` from external
```

Comment:
Why `<pyspark.sql.DataFrame>` is needed?

Reply:
Writing `` :class:`DataFrame` `` results in documentation that does not link to the class, and makes Sphinx output the following warning when running Sphinx in nit-picky mode (`-n` on the command line or `nitpicky = True` in the Sphinx config):
> spark/python/pyspark/sql/streaming.py:docstring of pyspark.sql.streaming.DataStreamReader:1: WARNING: py:class reference target not found: DataFrame

One could also just write `` :class:`pyspark.sql.DataFrame` `` to fix the warning and documentation linkage, but then the docs are rendered with the full `pyspark.sql.DataFrame` name. I figured rendering only `DataFrame` by writing `` :class:`DataFrame <pyspark.sql.DataFrame>` `` was more fitting with the rest of the docs, but I can of course change this is you'd prefer.

Reply:
okie